docs/framework/chat-assistant.mdx

```diff
@@ -2,153 +2,325 @@
-  <Step title="Create a handler for the Chatbot component">
-    Next, create a handler for incoming messages by adding the `handle_simple_message` handler. This method will manage the chat interactions:
 
-    ```python
-    def handle_simple_message(state, payload):
-        state["conversation"] += payload
-        
-        for chunk in state["conversation"].stream_complete():
+  <Step title="Add a Text component to the Section">
+  Drag a **Text** component from the **Toolkit** panel on the left (it's under **Content**, and you may need to scroll down a little to find it) and into the *Section*. Sections can act as containers for other components.
+
+  Select the **Text** component. In the **Properties** panel, set the **Text** property to provide instructions or context for your chat assistant. Here's an example: `Welcome to the Chat Assistant. Ask me anything!`
+  </Step>
+
+  <Step title="Add a Chatbot component to the Section">
+  The heart of this app is the **Chatbot** component, a pre-built component that displays the conversation between the LLM and the user and provides a text field where the user can enter prompts.
+
+  Drag a **Chatbot** component from the **Toolkit** panel (it's under **Content**) into the *Section*, just below the Text box.
+  </Step>
+</Steps>
+
+The project editor should now look like this:
+
+![Project editor with components laid out](/framework/images/tutorial/chat/chat_assistant_2c.png)
+
+
+## Add the back-end code
+
+With the UI laid out, it's time to work on the logic behind it. 
+
+Switch to your code editor and open the `main.py` file in your project's directory. This file was automatically generated; you'll update the code in it to define the behavior of your app.
+
+<Steps>
+  <Step title="Import libraries and load the Writer Framework API key">
+  You should see the following at the start of the file:
+
+  ```python
+  import writer as wf
+  import writer.ai
+  ```
+
+  Replace that code with the following:
+
+  ```python
+  import os
+  from dotenv import load_dotenv
+  import writer as wf
+  import writer.ai
+
+  # Load environment variables from .env file
+  load_dotenv()
+
+  # Set the API key
+  wf.api_key = os.getenv("WRITER_API_KEY")
+  ```
+  
+  This code imports the libraries that the application will need and then reads your Writer Framework API key from the `.env` file in your project's directory.
+  </Step>
+
+  <Step title="Initialize the application">
+  The next step is to set the application's initial state. Find this code...
+
+  ```python
+  # Welcome to Writer Framework! 
+  # This template is a starting point for your AI apps.
+  # More documentation is available at https://dev.writer.com/framework
+
+
+  # Initialise the state
+  wf.init_state({
+      "my_app": {
+          "title": "AI STARTER"
+      },
+  })
+  ```
+
+  ...and replace it with the following:
+
+  ```python
+  # Initialize the state
+  wf.init_state({
+      "conversation": writer.ai.Conversation(),
```

Comment:
It might be worth calling out here that you can pass configuration into this such as what model you want to use and then link to the AI module guide.